framework/src/modules/chain/synchronizer/fast_chain_switching_mechanism.js

```diff
@@ -133,6 +133,44 @@ class FastChainSwitchingMechanism extends BaseSynchronizer {
 		return delegateList.includes(receivedBlock.generatorPublicKey);
 	}
 
+	/**
+	 * Request blocks from `fromID` ID to `toID` ID from an specific peer `peer`
+	 *
+	 * @param {object} peerId - The ID of the peer to target
+	 * @param {string} fromId - The starting block ID to fetch from
+	 * @param {string} toId - The ending block ID
+	 * @return {Promise<Array<object>>}
+	 */
+	async _requestBlocksWithinIDs(peerId, fromId, toId) {
+		const maxFailedAttempts = 10; // TODO: Probably expose this to the configuration layer?
+		const blocks = [];
+		let failedAttempts = 0; // Failed attempt === the peer doesn't return any block or there is a network failure (no response or takes too long to answer)
+		let lastFetchedID = fromId;
+
+		while (failedAttempts < maxFailedAttempts) {
+			const { data } = await this.channel.invoke('network:requestFromPeer', {
+				procedure: 'getBlocksFromId',
+				peerId,
+				data: {
+					blockId: lastFetchedID,
+				},
+			}); // Note that the block matching lastFetchedID is not returned but only higher blocks.
+
+			if (data && data.length) {
+				blocks.push(...data); // `data` is an array of blocks.
```

Comment:
I would replace this comment for a meaningful variable name

Reply:
Done! :) Thanks for the suggestion, makes sense.